src/serial/frame_format_id.cpp

```diff
@@ -0,0 +1,81 @@
+#include "adm/serial/frame_format_id.hpp"
+#include <sstream>
+#include "adm/detail/id_parser.hpp"
+
+namespace adm {
+
+  // ---- Operators ---- //
+  bool FrameFormatId::operator==(const FrameFormatId& other) const {
+    bool equal{get<FrameIndex>() == other.get<FrameIndex>()};
+    equal = equal && (has<ChunkIndex>() == other.has<ChunkIndex>());
+    if (equal && has<ChunkIndex>()) {
+      equal = (get<ChunkIndex>() == other.get<ChunkIndex>());
+    }
+    return equal;
+  }
+
+  bool FrameFormatId::operator!=(const FrameFormatId& other) const {
+    return !(*this == other);
+  }
+
+  bool FrameFormatId::operator<(const FrameFormatId& other) const {
+    return formatId(*this) < formatId(other);
+  }
```

Comment:
Perhaps use `OptionalComparator` for these?

Reply:
0c2051e4753b11dc72ddf7256939a63b8df2f4a9